analyzer/block/block.go

```diff
@@ -337,6 +354,10 @@ func (b *blockBasedAnalyzer) Start(ctx context.Context) {
 			// in the batch will exceed the current lock expiry of 5min. The analyzer will terminate
 			// the batch early and attempt to refresh the locks for a new batch.
 			if b.slowSync {
+				// Only update this metric on slow-sync mode.
+				if err := b.sendQueueLengthMetric(ctx, height); err != nil {
+					b.logger.Warn("error updating queue length metric: %w", err)
+				}
```

Comment:
could put this in a separate goroutine and fetch both the latest processed block and latest chain block from the db

Reply:
This metric will actually be especially valuable during fast-sync, which is when the queue is of a non-trivial size. We can report an approximate value during fast-sync, that's fine. (Your docstring for QueueLength already already foresees this.)

We could fetch the latest node block at the beginning of the batch (around where we call fetchBatchForProcessing()), store it in a local variable, and update the metric at the end of `if err := b.processor.ProcessBlock() { ... }`. This ties into my comment about `processor::QueueLength()` probably not being needed.

Reply:
Hmm good point. Updated to your suggestion - with an additional tweak of also fetching the `nodeHeight` at each loop during slow-sync